exporters/zipkin/src/test/java/io/opentelemetry/exporter/zipkin/OtelToZipkinSpanTransformerTest.java

```diff
@@ -149,19 +152,194 @@ void generateSpan_ResourceServiceNameMapping() {
   void generateSpan_defaultResourceServiceName() {
     SpanData data = spanBuilder().setResource(Resource.empty()).build();
 
-    Endpoint expectedEndpoint =
+    Endpoint expectedLocalEndpoint =
         Endpoint.newBuilder()
             .serviceName(Resource.getDefault().getAttribute(ResourceAttributes.SERVICE_NAME))
             .ip(localIp)
             .build();
     Span expectedZipkinSpan =
         zipkinSpan(Span.Kind.SERVER, localIp).toBuilder()
-            .localEndpoint(expectedEndpoint)
+            .localEndpoint(expectedLocalEndpoint)
             .putTag(OtelToZipkinSpanTransformer.OTEL_STATUS_CODE, "OK")
             .build();
     assertThat(transformer.generateSpan(data)).isEqualTo(expectedZipkinSpan);
   }
 
+  @Test
+  void generateSpan_RemoteEndpointMapping() {
+    Attributes attributes =
+        Attributes.builder()
+            .put(PEER_SERVICE, "remote-test-service")
+            .put(NET_SOCK_PEER_ADDR, "8.8.8.8")
+            .put(NET_PEER_PORT, 42L)
+            .build();
+
+    SpanData spanData =
+        spanBuilder()
+            .setKind(SpanKind.CLIENT)
+            .setResource(Resource.empty())
+            .setAttributes(attributes)
+            .build();
+
+    Endpoint expectedLocalEndpoint =
+        Endpoint.newBuilder()
+            .serviceName(Resource.getDefault().getAttribute(ResourceAttributes.SERVICE_NAME))
+            .ip(localIp)
+            .build();
+
+    Endpoint expectedRemoteEndpoint =
+        Endpoint.newBuilder().serviceName("remote-test-service").ip("8.8.8.8").port(42).build();
+
+    Span expectedSpan =
+        zipkinSpan(Span.Kind.SERVER, localIp).toBuilder()
+            .kind(Span.Kind.CLIENT)
+            .localEndpoint(expectedLocalEndpoint)
+            .remoteEndpoint(expectedRemoteEndpoint)
+            .putTag(PEER_SERVICE.getKey(), "remote-test-service")
+            .putTag(NET_SOCK_PEER_ADDR.getKey(), "8.8.8.8")
+            .putTag(NET_PEER_PORT.getKey(), "42")
+            .putTag(OtelToZipkinSpanTransformer.OTEL_STATUS_CODE, "OK")
+            .build();
+
+    assertThat(transformer.generateSpan(spanData)).isEqualTo(expectedSpan);
+  }
+
+  @Test
```

Comment:
Nit: same here -- can we use `@ParameterizedTest` to check everything but `CLIENT` and `PRODUCER`?